src/panels/config/entities/ha-config-entities.ts

```diff
@@ -207,14 +210,23 @@ export class HaConfigEntities extends LitElement {
         title: "",
         label: localize("ui.panel.config.entities.picker.headers.state_icon"),
         type: "icon",
-        template: (entry) => html`
-          <ha-state-icon
-            title=${ifDefined(entry.entity?.state)}
-            slot="item-icon"
-            .hass=${this.hass}
-            .stateObj=${entry.entity}
-          ></ha-state-icon>
-        `,
+        template: (entry) =>
+          entry.icon
+            ? html`
+                <ha-state-icon
+                  title=${ifDefined(entry.entity?.state)}
+                  slot="item-icon"
+                  .hass=${this.hass}
+                  .stateObj=${entry.entity}
+                ></ha-state-icon>
+              `
+            : html`
+                <ha-icon
+                  icon=${until(
+                    entryIcon(this.hass, entry as EntityRegistryEntry)
```

Comment:
Should be cast as `EntityRegistryEntry` because `EntityRow` doesn't have `id` and `unique_id` but it's not needed for `entryIcon`.